src/shared/lib/encryption/encryption.ts

```diff
@@ -0,0 +1,110 @@
+import { Buffer } from 'buffer';
+
+import { IV_LENGTH } from '@shared/lib';
+
+import { crypto } from './crypto';
+
+type GetPrivateKeyProps = { userId: string; email: string; password: string };
+type GetPublicKeyProps = {
+  privateKey: number[];
+  appletPrime: number[];
+  appletBase: number[];
+};
+
+type GetAESKeyProps = {
+  userPrivateKey: number[];
+  appletPublicKey: number[];
+  appletPrime: number[];
+  appletBase: number[];
```

Comment:
Let's not bind to the "applet" and "user" name. Please keep them without it, e.g. userPrivateKey => privateKey, appletPublicKey => publicKey, etc.

That concerns type GetPublicKeyProps as well